databuilder/codes.py

```diff
@@ -55,9 +55,30 @@ class SNOMEDCTCode(BaseCode, system_id="snomedct"):
     "SNOMED CT"
 
 
-@dataclasses.dataclass(frozen=True)
+@dataclasses.dataclass()
 class Codelist:
-    codes: frozenset
+    codes: set[BaseCode]
+    category_maps: dict[str, dict[BaseCode, str]]
+
+    def __post_init__(self):
+        # Check that the same codes are used everywhere
+        for category_map in self.category_maps.values():
+            assert self.codes == category_map.keys()
+        # As a convenience we set every "reasonably" named categorisation as an
+        # attribute on the codelist so you can write e..g
```

Comment:
...

Reply:
I improved the comment here to better express my intentions.